### PR TITLE
Restore transformers tests and disable some tests

### PR DESCRIPTION
**Pull request by wangyems**
Temporarily disabled:
transformers/test_attention_fusion.py::TestFusion::test_gpt2_attention_fusion
transformers/test_optimizer.py::TestBertOptimization::test_gpt2_past_fp16
transformers/test_optimizer.py::TestBertOptimization::test_gpt2_past_mask

Will take a look and send a separate PR

### CODE REVIEW

File: tools/ci_build/requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # packages used by transformers tool test
-
+protobuf==3.17.0
```
Comment (snnn) [inline tools/ci_build/requirements.txt:2]: Why this change is needed?

Comment (wangyems) [reply tools/ci_build/requirements.txt:2]: As we sync last time, setting protobuf version will help with pipeline broke based on https://github.com/microsoft/onnxruntime/tree/snnn/p23

**Review by snnn — APPROVED** [review]
